parmed/gromacs/gromacstop.py

```diff
@@ -1422,6 +1423,19 @@ def write(self, dest, combine=None, parameters='inline'):
                               atom_type.mass, atom_type.charge, atom_type.sigma/10,
                               atom_type.epsilon*econv))
             parfile.write('\n')
+            # Nonbonded parameters
+            if self.has_NBFIX():
+                typemap = dict(self.parameterset.nbfix_types)
+                dest.write('[ nonbond_params ]\n')
+                for key, val in typemap.items():
+                    a1 = key[0]
+                    a2 = key[1]
+                    eps = val[0] # kcal
+                    sig = val[1] # Angstrom
+                    eps *= u.kilocalorie.conversion_factor_to(u.kilojoule)
```

Comment:
Calculate this value outside the loop so it doesn’t keep getting recalculated inside it.

Reply:
~~I'm not sure it is being calculated multiple times. Each unique pair of atom types has a `sig` and `eps` associated with them, and each of these cross interactions should only be interacted with once. Unless `self.parameterset.nbfix_types` has a key for each atom and not atom type.~~ Nevermind, I see you're referring to the conversion factor. I think.